client/my-sites/stats/site.jsx

```diff
@@ -233,26 +251,33 @@ class StatsSite extends Component {
 		if ( isDateControlEnabled ) {
 			// Sort out end date for chart.
 			const chartEnd = this.getValidDateOrNullFromInput( context.query?.chartEnd );
+
 			if ( chartEnd ) {
 				customChartRange = { chartEnd };
 			} else {
 				customChartRange = { chartEnd: moment().format( 'YYYY-MM-DD' ) };
 			}
+
 			// Sort out quantity for chart. Default to 30 days.
-			let daysInRange = 30;
+			let daysInRange = this.getDefaultRangeForPeriod( period );
 			const chartStart = this.getValidDateOrNullFromInput( context.query?.chartStart );
 			const isSameOrBefore = moment( chartStart ).isSameOrBefore( moment( chartEnd ) );
+
 			if ( chartStart && isSameOrBefore ) {
 				// Add one to calculation to include the start date.
 				daysInRange = moment( chartEnd ).diff( moment( chartStart ), 'days' ) + 1;
 				customChartRange.chartStart = chartStart;
 			} else {
+				// if start date is missing let the frequency of data take over to avoid showing one bar
+				// (e.g. months defaulting to 30 days and showing one point)
 				customChartRange.chartStart = moment()
 					.subtract( daysInRange, 'days' )
 					.format( 'YYYY-MM-DD' );
 			}
-			this.state.customChartQuantity = quantityForDaysAndPeriod( daysInRange, period );
-			customChartRange.daysInRange = daysInRange;
+
+			// how many units of data to ask for the bar for a given period
+			this.state.customChartQuantity = convertDaysToPeriodUnits( daysInRange, period );
+			customChartRange.daysInRange = daysInRange; // never used
```

Comment:
@a8ck3n do you recall if this is used anywhere? I can't find any reference at the moment.

Reply:
It's used when matching the current shortcut here:

https://github.com/Automattic/wp-calypso/blob/a5bd4ee8e3f589419c6cf9ad6e3d11a36b0cb992/client/components/stats-date-control/index.tsx#L106